### PR TITLE
Document arguments that use tidyselect

### LINKED ISSUE #872: Use tidy select labels like in dplyr

### PR DESCRIPTION
**Pull request by hadley**
And update to tidyselect 1.0.0 interface (doing the minimal to get it working again; not thinking through how to handle data frames with duplicate names).

Fixes #872

### CODE REVIEW

File: R/doc-params.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#' @description
+#' This page the describes the `<tidy-select>` argument modifier which indicates
+#' the argument uses tidy evaluation with **tidy selection**. Tidy selection
+#' provides a concise DSL for selecting variables based on their names.
+#' If you've never head of tidy evaluation before, start with
+#' `vignette("programming")`.
+#'
+#' # General usage
+#'
+#' If you have a data frame with variables `apple`, `banana`, `cantelope`,
```
Comment (jennybc) [inline R/doc-params.R:12]:
```suggestion
#' If you have a data frame with variables `apple`, `banana`, `cantaloupe`,
```
There are cantaloupes 🍈 and then there are antelopes 🦌

File: R/doc-params.R [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#'
+#' * Select data-variables stored in a env-variable with `all_of()` (which
+#'   will error if a variable is not found) or `any_of()` (which is
+#'   relaxed and will silently drop missing variables), e.g.
+#'   if `vars <- c("apple", "fig", "peach")`, then `all_of(vars)` will
+#'   error; `any_of(vars)` will select `apple` and `fig`.
+#'
+#' * Select contiguous variables with `:`, e.g. `apple:date`.
+#'
+#' * Select variables with named-based helpers: e.g. `ends_with("a")`,
```
Comment (jennybc) [inline R/doc-params.R:25]:
```suggestion
#' * Select variables with name-based helpers: e.g. `ends_with("a")`,
```

File: R/doc-params.R [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#'   `contains("g")`. See full list in [tidyselect::select_helpers].
+#'
+#' * Select variables of a given type with a `is` function: `is.numeric`,
```
Comment (jennybc) [inline R/doc-params.R:28]:
```suggestion
#' * Select variables of a given type with an `is` function: `is.numeric`,
```

File: R/doc-params.R [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+#' # Indirection
+#'
+#' There are two main cases:
+#'
+#' *   If you have a character vector of column names, use `all_of()`
+#'     or `any_of()`, depending on whether or not you want unknown variable
+#'     names to cause an error, e.g `unnest(df, all_of(vars))`,
+#'     `unnest(df, -any_of(vars))`.
+#'
+#' *   If you you want the user to supply a tidyselect specification in a
```
Comment (jennybc) [inline R/doc-params.R:48]:
```suggestion
#' *   If you want the user to supply a tidyselect specification in a
```

File: R/doc-params.R [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+#'     function argument, embrace the function argument, e.g
```
Comment (lionel-) [inline R/doc-params.R:49]: in a function argument, you need to tunnel the selection through the function argument. This is done by embracing the function argument with `{{`, e.g. `unnest(df, {{ vars }})`.

File: R/doc-params.R [hunk 6]
```diff
@@ -0,0 +1,54 @@
+#' Argument type: tidy-select
+#'
+#' @description
+#' This page the describes the `<tidy-select>` argument modifier which indicates
+#' the argument uses tidy evaluation with **tidy selection**. Tidy selection
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/doc-params.R]: I'll tweak it a bit but I do want to explicitly make the connection that tidy selection is a subcategory of tidy evaluation

**Pull request merged**